### PR TITLE
[bugfix] Fix Go Language Detection when the Process Check is Disabled

### PR DESCRIPTION
**Pull request by just-chillin**
### What does this PR do?

This PR fixes a bug where the language detection module does not query the system probe when using the process collector. This bug occured due to the fact that the core config was passed to the workload meta extractor instead of the system probe config, which the language detection module uses to know whether the system probe's language detection module is enabled.

### Motivation

Bugfix

### Possible Drawbacks / Trade-offs

None

### Describe how to test/QA your changes

```yaml
# datadog.yaml
language_detection: { enabled: true }
```
```yaml
# system-probe.yaml
system_probe_config:
  language_detection: { enabled: true } 
```
Using the above config:
1. Run the agent
2. Wait 1 minute for the first process collector run to occur
3. Run `agent workload-list`, and look for at least one go process. The agent is written in go, so it should be able to detect itself.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/events/monitor.go [hunk 1]
```diff
@@ -118,7 +116,7 @@ func (e *eventMonitor) UnregisterHandler(handler ProcessEventHandler) {
 	e.Lock()
 	defer e.Unlock()
 
-	if idx := slices.Index(e.handlers, handler); idx >= 0 {
-		e.handlers = slices.Delete(e.handlers, idx, idx+1)
-	}
+	//if idx := slices.Index(e.handlers, handler); idx >= 0 {
```
Comment (hmahmood) [inline pkg/network/events/monitor.go:119]: Should be removed?

**Review by hmahmood — APPROVED** [review]

File: pkg/network/events/monitor.go (hunk 1 above)
Comment (just-chillin) [reply pkg/network/events/monitor.go:119]: Oh yep, thanks didn't mean to commit that - was having compile errors on my machine

File: pkg/network/events/monitor.go
Comment (robertjli) [inline pkg/network/events/monitor.go:1]: Why was this file changed? I see that one of the imports is removed, but no code was changed?

Comment (robertjli) [reply pkg/network/events/monitor.go:1]: Ah, I see Hasan's previous comment. Can you just revert the whole file?

Comment (just-chillin) [reply pkg/network/events/monitor.go:1]: Yep, sorry thought I got it all 🙇🏻‍♀️

**Review by robertjli — APPROVED** [review]
